pkg/backup/backup_new_test.go

```diff
@@ -1629,6 +1631,280 @@ func TestBackupWithSnapshots(t *testing.T) {
+						mock.Anything, // pod
+						"ns-2",        // pod namespace
+						"pod-2",       // pod name
+						"hook-1",      // hook name
+						&velerov1.ExecHook{
+							Command: []string{"ls", "/tmp"},
+						},
+					},
+					returnArguments: []interface{}{nil},
+				},
+			},
+			wantBackedUp: []string{
+				"resources/pods/namespaces/ns-1/pod-1.json",
+				"resources/pods/namespaces/ns-2/pod-2.json",
+			},
+		},
+		{
+			name: "post hook with no resource filters runs for all pods",
+			backup: defaultBackup().
+				Hooks(velerov1.BackupHooks{
+					Resources: []velerov1.BackupResourceHookSpec{
+						{
+							Name: "hook-1",
+							PostHooks: []velerov1.BackupResourceHook{
+								{
+									Exec: &velerov1.ExecHook{
+										Command: []string{"ls", "/tmp"},
+									},
+								},
+							},
+						},
+					},
+				}).
+				Backup(),
+			apiResources: []*apiResource{
+				pods(
+					newPod("ns-1", "pod-1"),
+					newPod("ns-2", "pod-2"),
+				),
+			},
+			wantExecutePodCommandCalls: []*expectedCall{
+				{
+					arguments: []interface{}{
+						mock.Anything, // logger
+						mock.Anything, // pod
+						"ns-1",        // pod namespace
+						"pod-1",       // pod name
+						"hook-1",      // hook name
+						&velerov1.ExecHook{
+							Command: []string{"ls", "/tmp"},
+						},
+					},
+					returnArguments: []interface{}{nil},
+				},
+				{
+					arguments: []interface{}{
+						mock.Anything, // logger
+						mock.Anything, // pod
+						"ns-2",        // pod namespace
+						"pod-2",       // pod name
+						"hook-1",      // hook name
+						&velerov1.ExecHook{
+							Command: []string{"ls", "/tmp"},
+						},
+					},
+					returnArguments: []interface{}{nil},
+				},
+			},
+			wantBackedUp: []string{
+				"resources/pods/namespaces/ns-1/pod-1.json",
+				"resources/pods/namespaces/ns-2/pod-2.json",
+			},
+		},
+		{
+			name: "item is not backed up if hook returns an error when OnError=Fail",
+			backup: defaultBackup().
+				Hooks(velerov1.BackupHooks{
+					Resources: []velerov1.BackupResourceHookSpec{
+						{
+							Name: "hook-1",
+							PreHooks: []velerov1.BackupResourceHook{
+								{
+									Exec: &velerov1.ExecHook{
+										Command: []string{"ls", "/tmp"},
+										OnError: velerov1.HookErrorModeFail,
+									},
+								},
+							},
+						},
+					},
+				}).
+				Backup(),
+			apiResources: []*apiResource{
+				pods(
+					newPod("ns-1", "pod-1"),
+					newPod("ns-2", "pod-2"),
+				),
+			},
+			wantExecutePodCommandCalls: []*expectedCall{
+				{
+					arguments: []interface{}{
+						mock.Anything, // logger
+						mock.Anything, // pod
+						"ns-1",        // pod namespace
+						"pod-1",       // pod name
+						"hook-1",      // hook name
+						&velerov1.ExecHook{
+							Command: []string{"ls", "/tmp"},
+							OnError: velerov1.HookErrorModeFail,
+						},
+					},
+					returnArguments: []interface{}{errors.New("exec hook error")},
+				},
+				{
+					arguments: []interface{}{
```

Comment:
Given these arguments are pretty similar across all these tests, couldn't we omit the first two args and last one and just build them in the test code? Or is it possible other tests could change these args?

Reply:
Yeah, I think that's reasonable - I'll tweak next week.

Reply:
I did a little refactoring here which I think addresses your comment, plus added one more test case for the pre+post hook scenario. If it looks good, please squash + merge!

Reply:
Thanks, this looks a lot simpler!